androidHyperskillApp/src/main/java/org/hyperskill/app/android/home/view/ui/fragment/HomeFragment.kt

```diff
@@ -97,17 +101,23 @@ class HomeFragment :
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
-        setProblemsLimitFragment()
         initViewStateDelegate()
         initGamificationToolbarDelegate()
         problemOfDayCardFormDelegate.setup(viewBinding.homeScreenProblemOfDayCard)
         topicsToDiscoverNextDelegate.setup(
             requireContext(),
             viewBinding.homeTopicsToDiscoverNext.homeTopicsToDiscoverNextRecycler
         )
+        problemsLimitDelegate = ProblemsLimitDelegate(
+            viewBinding = viewBinding.homeProblemsLimit,
+            onNewMessage = {
+                homeViewModel.onNewMessage(HomeFeature.Message.ProblemsLimitMessage(it))
+            }
+        )
+        problemsLimitDelegate?.setup(context = requireContext())
         with(viewBinding) {
```

Comment:
Let's extract it as a separate function, like a `initGamificationToolbarDelegate`